docker-app/qfieldcloud/subscription/tests/test_subscription.py

```diff
@@ -724,6 +724,41 @@ def test_create_default_plan_subscription_raises_when_subscription_already_exist
         with self.assertRaises(django.db.utils.IntegrityError):
             Subscription.create_default_plan_subscription(u1.useraccount)
 
+    def test_remaining_trial_organizations_is_set_and_decremented(
+        self,
+    ):
+        user_plan = Plan.objects.get(code="default_user")
+        user_plan.max_trial_organizations = 2
+        user_plan.save(update_fields=["max_trial_organizations"])
+        u1 = Person.objects.create(username="u1")
+
+        # remaining_trial_organizations is set on create_subscription
+        self.assertEqual(u1.remaining_trial_organizations, 2)
+
+        trial_plan = Plan.objects.get(code="default_org")
+        trial_plan.is_trial = True
+        trial_plan.save(update_fields=["is_trial"])
+
+        # remaining_trial_organizations is decremented when creating a trial organization
+        self.assert_create_org_decrements_count("org1", u1, 1)
+
+        # remaining_trial_organizations is decremented down to 0
+        self.assert_create_org_decrements_count("org2", u1, 0)
+
+        # It doesn't directly prevent creating more trials, is just a counter that stays at 0
+        self.assert_create_org_decrements_count("org3", u1, 0)
+
+    def assert_create_org_decrements_count(
+        self, org_name, user, remaining_trial_organizations
+    ):
+        Organization.objects.create(
+            username=org_name, organization_owner=user, created_by=user
+        )
+        user.refresh_from_db()
+        self.assertEqual(
+            user.remaining_trial_organizations, remaining_trial_organizations
+        )
```

Comment:
```suggestion
        def assert_create_org_decrements_count(
	        org_name, user, remaining_trial_organizations
	    ):
	        Organization.objects.create(
	            username=org_name, organization_owner=user, created_by=user
	        )
	        user.refresh_from_db()
	        self.assertEqual(
	            user.remaining_trial_organizations, remaining_trial_organizations
	        )

        # remaining_trial_organizations is decremented when creating a trial organization
        assert_create_org_decrements_count("org1", u1, 1)

        # remaining_trial_organizations is decremented down to 0
        assert_create_org_decrements_count("org2", u1, 0)

        # It doesn't directly prevent creating more trials, is just a counter that stays at 0
        assert_create_org_decrements_count("org3", u1, 0)
```

If you prefer to have the function, then define it withing the test. I think this is safer and easier to read. Ideally I usually avoid having such "helper" functions, as they may cause trouble in the future when reading. But in this case I think it is ok.

Reply:
OK, thanks - will remember for future